### PR TITLE
FormLayout: added "help text" to a form text field

### PR DESCRIPTION
**Pull request by jeroendesloovere**
## Type

- Feature

## Pull request description

> Allows you to add "help text" to a form text field.

Adding the attribute help, will now create a ".help-block" class span just below the text field.
```
$builder->add(
    'name',
    TextType::class,
    [
        'label' => 'lbl.Naming',
        'attr' => [
            'help' => 'msg.NamingHelp',
        ],
    ]
)
```
Before:
schermafbeelding 2017-08-24 om 11 10 27

After:
schermafbeelding 2017-08-24 om 11 07 48

Updated: You can also use the following in your template:
```
{{ form_widget(form.name, {'attr.help': 'msg.NamingHelp'|trans}) }}
```
> Which will do the same thing.

### CODE REVIEW

**Comment by jeroendesloovere** [comment]
Note: when using bootstrap 4, the code should be `<span class="form-text text-muted">...</span>`

**Comment by jeroendesloovere** [comment]
Change added: you can now also use `{{ form_widget(form.name, {'help': 'msg.NamingHelp'|trans}) }}`

File: src/Backend/Core/Layout/Templates/FormLayout.html.twig [hunk 1]
```diff
@@ -374,3 +380,13 @@
     {% endif %}
   </div>
 {% endblock %}
+
+{% block help %}
+  {% for attributeName, attributeValue in attr %}
+    {% if attributeName == 'help' %}<span class="help-block">{{ attributeValue|trans }}</span>{% endif %}
```
Comment (carakas) [inline src/Backend/Core/Layout/Templates/FormLayout.html.twig]: you can now have 2 help blocks by doing it like this if you set one from the template and one from the php can you fix this please, also if possible, avoid the duplicate html

Comment (jeroendesloovere) [reply src/Backend/Core/Layout/Templates/FormLayout.html.twig]: Yes indeed, that is possible.
Is that a problem?

Comment (carakas) [reply src/Backend/Core/Layout/Templates/FormLayout.html.twig]: why would you even allow that?
I don't mind that you can overwrite it from the template, but this is a very strange implementation

Comment (jeroendesloovere) [reply src/Backend/Core/Layout/Templates/FormLayout.html.twig]: So you prefer the "attr.help"? And not the variable "help"?

Comment (carakas) [reply src/Backend/Core/Layout/Templates/FormLayout.html.twig]: indeed

**Comment by WouterSioen** [comment]
nice feature, I like

**Pull request merged**